sources/web/datalab/static/datalab.js

```diff
@@ -12,13 +12,17 @@
  * the License.
  */
 
+var debug = {
+  enabled: true,
+  log: function() { console.log.apply(console, arguments); }
+};
+
+function shimWebSockets() {
+    return location.host.substr(-12) === '.appspot.com';
```

Comment:
I wanted to avoid 'endsWith' because its relatively new and I was worried about users with older browsers (e.g. versions of IE prior to IE 12)